File: generated-sources/scala-finch/mojang-sessions/src/main/scala/org/openapitools/models/PlayerProfileTexturePropertyValue.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import org.openapitools.models.PlayerTexture

/**
 * This is obtained by decoding the \"textures\" property
 * @param timestamp UNIX timestamp in milliseconds
 * @param profileId The player UUID without hyphens
 * @param profileName The player's name
 * @param signatureRequired Only present if unsigned was set to false in the request
 * @param textures 
 */
case class PlayerProfileTexturePropertyValue(timestamp: Long,
                profileId: String,
                profileName: String,
                signatureRequired: Option[Boolean],
                textures: Option[PlayerTexture]
                )

object PlayerProfileTexturePropertyValue {
    /**
     * Creates the codec for converting PlayerProfileTexturePropertyValue from and to JSON.
     */
    implicit val decoder: Decoder[PlayerProfileTexturePropertyValue] = deriveDecoder
    implicit val encoder: ObjectEncoder[PlayerProfileTexturePropertyValue] = deriveEncoder
}

File: generated-sources/php-laravel/mojang-sessions/lib/app/Models/PlayerProfileTexturePropertyValue.php
<?php
/**
 * PlayerProfileTexturePropertyValue
 */
namespace app\Models;

/**
 * PlayerProfileTexturePropertyValue
 */
class PlayerProfileTexturePropertyValue {

    /** @var int $timestamp UNIX timestamp in milliseconds*/
    private $timestamp;

    /** @var string $profileId The player UUID without hyphens*/
    private $profileId;

    /** @var string $profileName The player&#39;s name*/
    private $profileName;

    /** @var bool $signatureRequired Only present if unsigned was set to false in the request*/
    private $signatureRequired;

    /** @var \app\Models\PlayerTexture $textures */
    private $textures;

}
